python/tests/wrapper/test_01_synthetic_collection.py

```diff
@@ -0,0 +1,148 @@
+# This example generates a synthetic in-memory collection with 100 items and 60 tokens.
+# First 40 tokens are split on 10 clear topics according to this condition: "(token_id % 10) == (item_id % 10)"
+# Last 20 tokens are randomly allocated to documents.
+# The example demonstrates that in this clear situation BigARTM can precisely recover all topics.
+
+import artm.wrapper
+import artm.wrapper.messages_pb2 as messages
+import artm.wrapper.constants as constants
+import random
+import uuid
+import shutil
+import os
+import tempfile
+import itertools
+import pytest
+
+def test_func():
+    # Set some constants
+    num_tokens = 60
+    num_items = 100
+    batches_folder = tempfile.mkdtemp()
+    pwt = 'pwt'
+    nwt = 'nwt'
+    
+    num_topics = 10
+    inner_iterations_count = 10
+    outer_iterations_count = 10
+    num_top_tokens = 4
+
+    perplexity_tol = 3.0
+    expected_perplexity_value_on_iteration = {
+        0: 55.0,
+        1: 37.0,
+        2: 27.5,
+        3: 22.5,
+        4: 21.0,
+        5: 20.5,
+        6: 20.5,
+        7: 20.5,
+        8: 20.5,
+        9: 20.5,
+    }
+
+    top_tokens_tol = 0.2
+    expected_top_tokens_weight = 0.1
+
+    # Generate small collection of random items
+    batch = messages.Batch()
+    batch.id = str(uuid.uuid4())
+    for token_id in xrange(num_tokens):
+        batch.token.append('token_{0}'.format(token_id))
+
+    for item_id in xrange(num_items):
+        item = batch.item.add()
+        item.id = item_id
+        field = item.field.add()
+        for token_id in xrange(num_tokens):
+            field.token_id.append(token_id)
+            background_count = random.randint(1, 5) if (token_id >= 40) else 0
```

Comment:
Never use random in tests, it makes them stochastic.

Imagine the situation: you receive a build fail, tried to change the code to fix the potential problem but don't fix them actually, then you re-run tests and they are ok because of happiness of random number! You don't fix the problem, but CI system says you are.

At least you should set the random seed: `random.seed(constant_number)`. It is better to wrap your test in a parametrized function `test01(background_count)` and run this test for several fixed argument values.